### PR TITLE
fix: Fix missing dependent repos and missing packages in cmake

### PR DESCRIPTION
**Pull request by jerry73204**
I ran a clean build on [nvidia/cuda:11.7.1-cudnn8-devel-ubuntu20.04](https://hub.docker.com/layers/nvidia/cuda/11.7.1-cudnn8-devel-ubuntu20.04/images/sha256-69c8bb0d2ef7b4f2a246f49846de2c6cb54c2f605db4907433e800d787edbce7?context=explore) docker image and it failed with several missing dependencies. This PR adds several missing deps to fix this build.

Besides, the container used in the container installation doc ([link](https://autowarefoundation.github.io/autoware-documentation/main/installation/autoware/docker-installation/)) does not have TensorRT installed. It causes some of the binary builds to be skipped. I think the doc could mention that TensorRT must be manually installed.

### CODE REVIEW

**Comment by kenji-miyake** [comment]
@jerry73204 Thank you for the PR.

> I ran a clean build on [nvidia/cuda:11.7.1-cudnn8-devel-ubuntu20.04](https://hub.docker.com/layers/nvidia/cuda/11.7.1-cudnn8-devel-ubuntu20.04/images/sha256-69c8bb0d2ef7b4f2a246f49846de2c6cb54c2f605db4907433e800d787edbce7?context=explore) docker image and it failed with several missing dependencies.

Could you show me the commands you ran, please?
Also, please note that `CUDA 11.7` isn't officially supported yet.

> Besides, the container used in the container installation doc ([link](https://autowarefoundation.github.io/autoware-documentation/main/installation/autoware/docker-installation/)) does not have TensorRT installed.

I'm sorry, it's a bit old. You have to use the `latest-cuda` image.
I'll update the documentation.

**Comment by jerry73204** [comment]
> I'll update the documentation.

Thanks. Can you kindly tell the CUDA, cuDNN and TensorRT versions you're using?

For the build process, I did not have a Dockerfile but did docker-commit to save the image.
Here are the steps I can remember.

Launch a container with CUDA and cuDNN installed.

```bash
rocker --nvidia --x11 --user --volume $PWD -- nvidia/cuda:11.7.1-cudnn8-devel-ubuntu20.04
```

In the container shell, run these steps to build autoware.universe.

```bash
## Install build tools
sudo apt install git python3-vcstool python3-colcon-common-extensions python3-rosdep

## Follow the ROS galactic tutorial to configure the system.
## https://docs.ros.org/en/galactic/Installation/Ubuntu-Install-Debians.html
## Then,
sudo apt install ros-galactic-desktop

## Download and install TensorRT repo .deb from
## https://developer.nvidia.com/nvidia-tensorrt-8x-download
## Then,
sudo apt install tensorrt-dev


## Prepare autoware.universe repo and apply the patch in this PR.
cd autoware.universe
rm -rf src build install log  # make sure the build is clean
mkdir src && vcs import src < build_depends.repos
colcon build

## Finally, check that the build is successful
echo $?
```

**Comment by kenji-miyake** [comment]
> Thanks. Can you kindly tell the CUDA, cuDNN and TensorRT versions you're using?

@jerry73204 Sure, please refer to this file: https://github.com/autowarefoundation/autoware/blob/main/amd64.env#L5-L7

> For the build process, I did not have a Dockerfile but did docker-commit to save the image.
Here are the steps I can remember.

Thank you.
Have you confirmed the [official installation steps](https://autowarefoundation.github.io/autoware-documentation/main/installation/)?
If not, please check it out first. :bow:

File: build_depends.repos [hunk 1]
```diff
@@ -41,3 +41,19 @@ repositories:
...
+    url: https://github.com/tier4/cudnn_cmake_module.git
+    version: 0.0.1
+  universe/external/point_cloud_msg_wrapper:
+    type: git
+    url: https://gitlab.com/ApexAI/point_cloud_msg_wrapper.git
+    version: galactic
+  universe/external/tensorrt_cmake_module:
+    type: git
+    url: https://github.com/tier4/tensorrt_cmake_module.git
+    version: 0.0.3
```
Comment (kenji-miyake) [inline build_depends.repos:59]: I believe you should build `autoware.universe` with [autowarefoundation/autoware/autoware.repos](https://github.com/autowarefoundation/autoware/blob/main/autoware.repos).

**Review by kenji-miyake — CHANGES_REQUESTED** [review]

**Comment by mitsudome-r** [comment]
@jerry73204 Could you make the fix according to @kenji-miyake 's review?

**Comment by jerry73204** [comment]
I went through the instructions on a clean Ubuntu and it worked. My guess is that the ansible script scrambled the existing CUDA installation on the previous PC. My workaround was to remove all cuda/nvidia packages and clean up PATH and LD_LIBRARY_PATH in .profile and .bashrc, and clean up apt sources. Install the deps manually instead of ansible script and it worked.

Perhaps the doc could recommend not to use ansible script when cuda is already installed. It could save me hours.

**Comment by jerry73204** [comment]
I think it's a doc issue. The PR can be closed.

**Comment by jerry73204** [comment]
The PR is stale and is no longer needed.

**Pull request closed without merging**